apps/coordinator/src/actions/transactionActions.js

```
@@ -300,97 +309,219 @@ export function setMaxSpendOnOutput(outputIndex) {
+    const feeSats = inputsTotalSats.minus(outputsTotalSats);
     const fee = satoshisToBitcoins(feeSats);
     dispatch(setFee(fee));
+  };
+}
 
-    dispatch(finalizeOutputs(true));
+export function importPSBT(psbtText) {
+  return (dispatch, getState) => {
+    let state = getState();
+    const { network } = state.settings;
+    try {
+      // Handles both PSBTv0 and PSBTv2
+      const psbt = loadPsbt(psbtText, network);
+      if (!psbt) {
+        throw new Error("Could not parse PSBT.");
+      }
+
+      if (psbt.txInputs.length === 0) {
+        throw new Error("PSBT does not contain any inputs.");
+      }
+      if (psbt.txOutputs.length === 0) {
+        throw new Error("PSBT does not contain any outputs.");
+      }
+
+      dispatch(resetOutputs());
+      dispatch(setUnsignedPSBT(psbt.toBase64()));
+
+      // ==== PROCESS INPUTS ====
+      const inputs = selectInputsFromPSBT(getState(), psbt);
 
-    // In the future, if we want to support loading in signatures
-    // (or sets of signatures) included in a PSBT, we likely need to do
-    // that work here. Initial implementation just ignores any signatures
-    // included with the uploaded PSBT.
+      if (inputs.length === 0) {
+        throw new Error("PSBT does not contain any UTXOs from this wallet.");
+      }
+      if (inputs.length !== psbt.txInputs.length) {
+        throw new Error(
+          `Only ${inputs.length} of ${psbt.txInputs.length} PSBT inputs are UTXOs in this wallet.`,
+        );
+      }
+
+      dispatch(setInputs(inputs));
+
+      // ==== PROCESS INPUTS ====
+      const { outputsTotalSats } = dispatch(setOutputsFromPSBT(psbt));
+
+      // Calculate and set fee
+      dispatch(setFeeFromState(outputsTotalSats));
+
+      // ==== Extract and import signatures (If they are present)====
+      dispatch(setSignaturesFromPsbt(psbt));
+
+      // Finalize the transaction
+      dispatch(finalizeOutputs(true));
+    } catch (error) {
```

this is superfluous imo. just let the error bubble up.

---

removed now .. thanks :)